auction-app-backend/src/main/java/com/example/auctionapp/service/CategoryService.java

```diff
@@ -0,0 +1,76 @@
+package com.example.auctionapp.service;
+
+import com.example.auctionapp.dto.request.CategoryRequestDTO;
+import com.example.auctionapp.dto.response.CategoryDTO;
+import com.example.auctionapp.entity.Category;
+import com.example.auctionapp.exceptions.repository.ResourceNotFoundException;
+import com.example.auctionapp.repository.CategoryRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class CategoryService {
```

Comment:
One important thing here, when we are doing this, we never expose Service like this. 
Standard approach is to have interface and implementation class. In this case let's say `CategoryService` will be interface with all needed methods and `CategoryServiceImpl` will be the class that implements the `CategoryService` interface and its methods. When you are injecting category service, you are using interface. We can discuss this more on the call, but for now please fix this.